_posts/2021-01-18-changelog-139.markdown

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #139"
+date: 2021-01-18 17:00:00 +0300
+---
+
+
+## New Features
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6497 -->
+* Take into account attributes under `#[cfg_attr]` attribute in our type inference, name resolution, and other analysis.
+  Now our code insight works much better with code containing attributes like `#[cfg_attr(..., derive(...))]`, `#[cfg_attr(..., path="...")]`, etc.
```

Comment:
```suggestion
  Now code insight works much better with code containing attributes like `#[cfg_attr(..., derive(...))]`, `#[cfg_attr(..., path="...")]`, etc.
```

Reply:
Let's avoid `we`, `our`, `us`, etc.

Reply:
Replaced
```
Such test attributes are quite complex to handle in the IDE because we do not want to conditionally disable (exclude from analysis and gray out) all the tests, but do want to conditionally disable only the mock objects
```
with
```
Such test attributes are quite complex to handle in the IDE: the tests should not be conditionally disabled since you might want to read and edit them at any time; but only mock objects should be grayed-out and excluded from code analysis
```
as well to get rid of personal construction